### PR TITLE
FEATURE: Add Discover in bottom tabs

### PR DESCRIPTION
**Pull request by pmusaraj**
A large PR, sorry in advance! 

This includes: 

- adding a bottom tabs navigation to make room for a Discover screen
- adding a Discover screen
- fixings links to group inboxes
- adjusting styling across the app a bit
- removing RNSVG dependency
- removes fallback for old notifications endpoint

### CODE REVIEW

File: js/ProgressBar.js [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 import React, {useContext} from 'react';
 import {Dimensions, View} from 'react-native';
-import {Bar} from 'react-native-progress';
+import Bar from 'react-native-progress/Bar';
```
Comment (pmusaraj) [inline js/ProgressBar.js:6]: This change makes the RNSVG dependency unnecessary.

File: js/site.js [hunk 2]
```diff
@@ -283,61 +278,8 @@ class Site {
...
       );
-      useApiFallbacks = true;
-    }
-
-    if (!useApiFallbacks) {
-      return 0;
-    }
-
-    // TODO(pmusaraj): remove after June 2024
-    // once most sites will have received the new API at /notifications/totals.json
```
Comment (pmusaraj) [inline js/site.js:296]: removes fallback for old notifications code via session.

File: js/screens/DiscoverScreen.js [hunk 3]
```diff
@@ -0,0 +1,431 @@
...
+      .then(json => {
+        if (json.topics) {
+          this.setState({
+            results: opts.append
+              ? this.state.results.concat(json.topics)
+              : json.topics,
+            loading: false,
+            hasMoreResults:
+              json.grouped_search_result &&
+              json.grouped_search_result.more_full_page_results,
```
Comment (CvX) [inline js/screens/DiscoverScreen.js:102]: 🏌️ 
```suggestion
              json.grouped_search_result?.more_full_page_results,
```

File: js/screens/DiscoverScreen.js [hunk 4]
```diff
@@ -0,0 +1,431 @@
...
+        }
+      })
+      .catch(e => {
+        console.log(e);
+      });
+  }
+
+  addSite(term, opts = {}) {
+    if (term.length === 0) {
+      return new Promise((resolve, reject) => reject());
```
Comment (CvX) [inline js/screens/DiscoverScreen.js:140]: Ideally all of it would consistently use `async/await` 😅

File: js/screens/HomeScreen.js [hunk 5]
```diff
@@ -159,94 +121,93 @@ class HomeScreen extends React.Component {
...
-          reject('failure');
-        })
-        .finally(() => {
-          setTimeout(() => {
-            this.setState({addSiteProgress: 0});
-          }, 1000);
-        });
-    });
-  }
+  // async promptForNotifications() {
```
Comment (CvX) [inline js/screens/HomeScreen.js:124]: Will we need this commented-out code?

File: js/site_manager.js [hunk 6]
```diff
@@ -533,11 +529,25 @@ class SiteManager {
...
 
+  connectedSitesCount() {
+    let count = 0;
+    this.sites.forEach(site => {
+      if (site.authToken) {
+        count++;
+      }
+    });
+
+    return count;
```
Comment (CvX) [inline js/site_manager.js:540]: 🏌️ 
```suggestion
    return this.sites.filter((site) => site.authToken).length;
```

File: js/locale/en.json [hunk 7]
```diff
@@ -1,7 +1,8 @@
 {
   "add_first_site": "+ Add your first site",
   "add_site": "Add Site",
-  "add_sites": "Add Discourse sites to keep track of",
+  "add_sites": "Get started following the Discourse communities you love!",
```
[the remark this answers is not in the document]
Comment (pmusaraj) [reply js/locale/en.json:4]: "Get started and follow..." sounds great, will go with that.

File: js/screens/DiscoverScreen.js (hunk 4 above)
Comment (pmusaraj) [reply js/screens/DiscoverScreen.js:138]: Yeah, true, this was moved here from a different place and I wanted to keep the changes to a minimum.

**Review by CvX — APPROVED** [review]

**Pull request merged**